gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilPlugin.kt

```diff
@@ -93,6 +103,58 @@ internal open class AnvilPlugin : KotlinCompilerPluginSupportPlugin {
         getConfiguration(target, "main").extendsFrom(commonConfiguration)
       }
     }
+
+    target.afterEvaluate {
```

Comment:
This isn't my favorite. Another alternative is we could maybe exchange the properties for a function that calls through to this setup synchronously

Reply:
We may need to do the synchronous approach actually. While testing this, I encoutnered this issue https://github.com/google/ksp/issues/1789. Alternatively, we could move to just using static gradle properties only

Reply:
nvm, I think this is probably enough for now, and advanced cases like ours can just apply ksp eagerly if need be

Reply:
The function would be my choice.  `afterEvaluate { }` is the worst.

We could still have Gradle properties, just not in the DSL.  Then the plugin could just explicitly call the function if they're toggled.

Reply:
alright let me see what I can do!

Reply:
Alright took a crack at it, how's this look? https://github.com/square/anvil/pull/901/commits/fc18e8c144c3f87856487663155554d8ee79f859

Reply:
Hmm, seems the deferred cli args breaks CC. Let me see if I can improve this

Reply:
Fixed in https://github.com/square/anvil/pull/901/commits/cf56054c9fa621cefd0a4527db440838f821b7df